earth_enterprise/docker/README.md

```diff
@@ -0,0 +1,169 @@
+
+```BASH
+docker pull thermopylae/opengee-experimental
+```
+
+
+## Running the Container
+
+### Running the Open GEE Servers
+
+#### Running Open GEE Server with Persistent Globe and Map Databases
+
+In order to be able to build and serve Earth databases that persist between 
+instantiations of the Docker image, you need a persistent assets directory 
+outside of the Docker container.  Another alternative is to run a container
+with a temporary assets directory, and commit the running to container to a
+new image after you've build your assets.
+
+To run the Docker container using a persistent globe assets directory outside 
+the container, use the following script.  Set the `HOST_GEVOL_PATH`
+environment variable to the path outside the container where you want to store
+your persistent assets.
+
+```BASH
+HOST_GEVOL_PATH=/persistent/gevol/ ./bin/start-docker-image.sh
+```
+
+You can optionally set the `GEE_SERVER_PORT` and even  `IMAGE_TAG` to override
+the default values.
+
+
+##### Initializing a New Location for Persistsing Databases
+
+If you want to create the directory structure needed for storing globe and map
+databases in a new location, you can use the following commands (after
+substituting the desired path), and follow the prompts.
+
+```BASH
+sudo mkdir -p /persistent/gevol/assets
+docker run --cap-add=DAC_READ_SEARCH -v /persistent/gevol:/gevol -ti opengee-experimental /opt/google/bin/geconfigureassetroot --new --assetroot /gevol/assets
+```
+
+
+#### Running Open GEE Server with Temporary Globe and Map Databases
+
+To run the Docker container using temporary assets that disapper when the 
+container is stopped, leave the `HOST_GEVOL_PATH` environment valiable empty.
+This is the default behavior
+
+```BASH
+HOST_GEVOL_PATH='' ./bin/start-docker-image.sh
+```
+
+
+#### Running the Fusion UI in the Docker Container
+
+To start the Fusion UI in the Open GEE Docker container, you need to have
+the GEE servers running in the container, e.g. by running
+`./bin/start-docker-image.sh`, and an X11 server outside the container that
+the Fusion UI can connect to.  If you started the container on a Linux 
+distribution running X11 listening for socket connections to `/tmp/.X11-unix`,
+you should only need to run the following command to bring up the Fusion UI:
+
+```BASH
+./bin/attach-fusion.sh
+```
+
+Running the Fusion UI on a non-Linux Docker host running X11 has not been
+tested, although it should be theoretially possible.
+
+
+#### Using GEE Tutorial Files in the Docker Container
+
+The Docker image is built without the GEE tutorial files, since people using
+the image for production may not want to carry the 600+ MB files they wouldn't
+use.
+
+##### Deriving an Image that Includes the Tutorial Files
+
+You can create an image based on the provided one that adds the tutorial
+files. All you need to do to construct the new image is to run:
+
+```Dockerfile
+CMD /bin/bash /opt/google/share/tutorials/fusion/download_tutorial.sh
```

Comment:
What did you try running?

I didn't include a complete code sample here.  I thought someone deriving a Docker image from another would already know what to do. 

There's more than one way to get the tutorial files in Docker.

For example, if you want to run a few tests, and you don't think you'll do much with the tutorial resources afterwards, you can just start a container from the published image, attach a BASH terminal and execute the above command.  You'll get the tutorial files, but when you shut down the container, they won't be kept.  So, if you need to use them again, you'll have to re-download them.

Another option would be to commit the running Docker container after executing the above command to download the tutorial resources.  The running image would be committed to a new image that contains the downloaded files.

One more possibility is to create a couple-line Docker file for an image based on the published GEE Docker image, and containing the download command above.  You could, then, build a new GEE image containing the tutorial files.

This seems more a matter of familiarity with Docker than anything very specific to GEE.  It may not be a bad idea to also publish an image including the tutorial files, since we may want people to be able to start learning how to use Fusion and GEE Server without have to know much about Docker.